src/control/cmd/daos_agent/start.go

```diff
@@ -95,17 +94,23 @@ func (cmd *startCmd) Execute(_ []string) error {
 	// Setup signal handlers so we can block till we get SIGINT or SIGTERM
 	signals := make(chan os.Signal)
 	finish := make(chan struct{})
-	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM)
 
+	signal.Notify(signals, syscall.SIGINT, syscall.SIGTERM, syscall.SIGPIPE)
 	// Anonymous goroutine to wait on the signals channel and tell the
-	// program to finish when it receives a signal. Since we only notify on
-	// SIGINT and SIGTERM we should only catch this on a kill or ctrl+c
+	// program to finish when it receives a signal. Since we notify on
+	// SIGINT and SIGTERM we should only catch these on a kill or ctrl+c
+	// SIGPIPE is caught and logged to avoid killing the agent.
 	// The syntax looks odd but <- Channel means wait on any input on the
 	// channel.
 	go func() {
 		sig := <-signals
-		cmd.log.Infof("Caught %s; shutting down", sig)
-		close(finish)
+		switch sig {
+		case syscall.SIGPIPE:
+			cmd.log.Infof("Signal received.  Caught non-fatal %s; continuing", sig)
```

Comment:
Is there some way we could grab a stack trace at this point and log that, too? Could be helpful in debugging.

Reply:
I put a stack dump routine in when I wrote the handler and put the output of that into the JIRA ticket for future reference.  I took it out of the proposed solution because we decided to handle SIGPIPE as 'normal' and not an 'error' condition.  Even though it could be logged to the debug log, I'm wondering if such a stack trace will look alarmist.  I'm fine either way if we get consensus to dump the trace.

Reply:
I don't feel strongly either way. Not sure how useful a stack trace at this point would be, but if you decide to add it I'd suggest logging it separately at debug. It would show up in CI logs, which is fine, but wouldn't appear for normal use.

Also, a small nit: "Signal received. Caught %s ..." is kind of redundant. The fact that it was caught and logged implies that it was received.